core/src/main/java/io/apicurio/hub/api/codegen/jaxrs/OpenApi2CodegenVisitor.java

```diff
@@ -335,107 +301,118 @@ public void visitRequestBody(Oas30RequestBody node) {
-                
+
                 clonedMethod.getArguments().add(cgArgument);
-                
+
                 return clonedMethod;
             }).collect(Collectors.toList());
             this._currentInterface.getMethods().remove(methodTemplate);
             this._currentInterface.getMethods().addAll(methods);
             this._currentMethods = methods;
         }
-        
     }
 
     /**
-     * @see io.apicurio.datamodels.combined.visitors.CombinedVisitorAdapter#visitResponse(io.apicurio.datamodels.openapi.models.OasResponse)
+     * @see io.apicurio.datamodels.models.openapi.v30.visitors.OpenApi30VisitorAdapter#visitResponse(io.apicurio.datamodels.openapi.models.OpenApiResponse)
      */
     @Override
-    public void visitResponse(OasResponse node) {
+    public void visitResponse(OpenApiResponse node) {
+        // Skip processing of response definitions (defined in /components/responses)
+        if (NodeUtil.isDefinition(node)) {
+            return;
+        }
+
+        String statusCode = getMappedNodeName(node);
         // Note: if there are multiple 2xx responses, only the first one will
         // become the method return value.
-        if (node.getStatusCode() != null && node.getStatusCode().indexOf("2") == 0 && this._currentMethods.get(0).getReturn() == null) {
-            if (node.ownerDocument().getDocumentType() == DocumentType.openapi2) {
-                this.visit20Response((Oas20Response) node);
-            }
-            if (node.ownerDocument().getDocumentType() == DocumentType.openapi3) {
-                this.visit30Response((Oas30Response) node);
-            }
-        }
-    }
-    private void visit20Response(Oas20Response node) {
-        if (node.getStatusCode() != null && node.getStatusCode().indexOf("2") == 0) {
-            this._currentMethods.forEach(_currentMethod -> _currentMethod.setReturn(this.returnFromSchema(node.schema)));
-        }
-    }
-    private void visit30Response(Oas30Response node) {
-        List<Oas30MediaType> mediaTypes = node.getMediaTypes();
-        // TODO if there are multiple response media types, handle it somehow - probably by returning a JAX-RS Response object
-        if (mediaTypes != null && mediaTypes.size() > 0) {
-            Oas30MediaType mediaType = mediaTypes.get(0);
-            Extension returnTypeExt = mediaType.getExtension(CodegenExtensions.RETURN_TYPE);
-            CodegenJavaReturn cgReturn = null;
-            if (returnTypeExt != null) {
-                String returnType = (String) returnTypeExt.value;
-                CodegenJavaReturn customReturn = new CodegenJavaReturn();
-                customReturn.setType(returnType);
-                cgReturn = customReturn;
-            } else {
-                cgReturn = this.returnFromSchema(mediaType.schema);
+        if (statusCode != null && statusCode.indexOf("2") == 0 && this._currentMethods.get(0).getReturn() == null) {
```

Comment:
Out of scope for this PR, but should it be `startsWith("2")` rather than `indexOf("2")` here?